src/static/helpers/customApi.ts

```diff
@@ -0,0 +1,151 @@
+    );
+  }
+  return foundHeader;
+};
+
+/**
+ * A helper function designed to make calls to a custom API endpoint
+ * For more information about custom APIs, please refer to the [API documentation](https://developer.salesforce.com/docs/commerce/commerce-api/guide/custom-apis.html)
+ * @param args - Argument object containing data used for custom API request
+ * @param args.options - An object containing any custom settings you want to apply to the request
+ * @param args.options.method? - The request HTTP operation. 'GET' is the default if no method is provided.
+ * @param args.options.parameters? - Query parameters that are added to the request
+ * @param args.options.customApiPathParameters? - Path parameters used for custom API. Required path parameters (apiName, endpointPath, organizationId, and shortCode) can be in this object, or args.clientConfig.parameters, where this object will take priority for duplicates. apiVersion is defaulted to 'v1' if not provided.
+ * @param args.options.headers? - Headers that are added to the request. Authorization header should be in this parameter or in the clientConfig.headers. If "Content-Type" is not provided in either header, it will be defaulted to "application/json".
+ * @param options.body? - Body that is used for the request. The body will be automatically formatted for Content-Type application/json and application/x-www-form-urlencoded
+ * @param args.clientConfig - Client Configuration object used by the SDK with properties that can affect the fetch call
+ * @param args.clientConfig.parameters - Path parameters used for custom API endpoints. The required properties are: apiName, endpointPath, organizationId, and shortCode. An error will be thrown if these are not provided.
+ * @param args.clientConfig.headers? - Additional headers that are added to the request. Authorization header should be in this argument or in the options?.headers. options?.headers will override any duplicate properties. If "Content-Type" is not provided in either header, it will be defaulted to "application/json".
+ * @param args.clientConfig.baseUri? - baseUri used for the request, where the path parameters are wrapped in curly braces. Default value is 'https://{shortCode}.api.commercecloud.salesforce.com/custom/{apiName}/{apiVersion}'
+ * @param args.clientConfig.fetchOptions? - fetchOptions that are passed onto the fetch request
+ * @param args.clientConfig.throwOnBadResponse? - flag that when set true will throw a response error if the fetch request fails (returns with a status code outside the range of 200-299 or 304 redirect)
+ * @param args.clientConfig.proxy? - Routes API calls through a proxy when set
+ * @param args.rawResponse? - Flag to return the raw response from the fetch call. True for raw response object, false for the data from the response
+ * @returns Raw response or data from response based on rawResponse argument from fetch call
+ */
+export const callCustomEndpoint = async (args: {
```

Comment:
Dumb question..and maybe this is a comment about helpers in general or static code. Would we not always have code duplication from `commerce-sdk` and `commerce-sdk-isomorphic`. E.g. this is the same code you have in that other lib. Meaning if there is a bug in it, you have to fix it in 2 places, not just one.

Reply:
Yes, this is correct. Unfortunately this is one of the gaps in having two separate SDKs. There were talks in the past for creating a core repo with code that is shared by both SDKs, but it wasn't ever followed up on.